### PR TITLE
making extended template work with v2 changes, adding it to taskcat

### PR DESCRIPTION
**Pull request by jvanbrie**
*Note: Please remember to review the [contribution guidelines](https://github.com/DataDog/cloudformation-template/blob/master/CONTRIBUTING.md)
if you have not yet done so.*

### What does this PR do?
Adds the recent resource collection and v2 changes to the extended template. Adds a way to test the extended template via taskcat.

### Testing Guidelines
Ran taskcat

### CODE REVIEW

File: aws_organizations/main_organizations.yaml [hunk 1]
```diff
@@ -330,7 +330,7 @@ Resources:
               reason = json.dumps(cfResponse)
               cfnresponse.send(
                   event,
-                  signalcontext,
+                  context,
```
Comment (jvanbrie) [inline aws_organizations/main_organizations.yaml:333]: Drive-by fix.

File: aws_quickstart/taskcat/run-taskcat-tests.sh [hunk 2]
```diff
@@ -31,7 +48,16 @@ for f in ../*.yaml; do
    sed "s|<BUCKET_PLACEHOLDER>.s3.amazonaws.com/aws/<VERSION_PLACEHOLDER>|${TASKCAT_S3_BUCKET}.s3.amazonaws.com/${TASKCAT_PROJECT}|g" $f > ./tmp/$(basename $f)
 done
 
-sed "s|<REPLACE_DD_API_KEY>|${DD_API_KEY}|g ; s|<REPLACE_DD_APP_KEY>|${DD_APP_KEY}|g ; s|<REPLACE_AWS_PROFILE>|${AWS_SSO_PROFILE_NAME}|g" ./.taskcat.yml > ./tmp/.taskcat.yml
+if [ "$TEST_VERSION" = "standard" ]; then
```
Comment (jvanbrie) [inline aws_quickstart/taskcat/run-taskcat-tests.sh:51]: I don't love the way I did this, but it works 🤷‍♂️.

**Review by ktmq — APPROVED** [review]
